instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java

```diff
@@ -23,7 +23,13 @@ public final class ServerSpan {
    */
   @Nullable
   public static Span fromContextOrNull(Context context) {
-    return SpanKey.SERVER.fromContextOrNull(context);
+    // try the HTTP semconv span first
+    Span span = SpanKey.HTTP_SERVER.fromContextOrNull(context);
+    // if it's absent then try the SERVER one - perhaps suppression by span kind is enabled
+    if (span == null) {
+      span = SpanKey.KIND_SERVER.fromContextOrNull(context);
+    }
+    return span;
```

Comment:
maybe we can generalize `ServerSpan` to `LocalRootSpan` to avoid this mix of semconv and kind-based span keys (not in this PR, I created #5857 to discuss)